src/nanovg.c

```diff
@@ -405,19 +408,25 @@ void nvgEndFrame(NVGcontext* ctx)
 			if (ctx->fontImages[i] != 0) {
 				int nw, nh;
 				nvgImageSize(ctx, ctx->fontImages[i], &nw, &nh);
-				if (nw < iw || nh < ih)
+				if (nw < iw || nh < ih) {
 					nvgDeleteImage(ctx, ctx->fontImages[i]);
-				else
+					ctx->fontImages[i] = 0;
+				} else {
 					ctx->fontImages[j++] = ctx->fontImages[i];
+				}
 			}
 		}
 		// make current font image to first
 		ctx->fontImages[j++] = ctx->fontImages[0];
 		ctx->fontImages[0] = fontImage;
 		ctx->fontImageIdx = 0;
 		// clear all images after j
-		for (i = j; i < NVG_MAX_FONTIMAGES; i++)
-			ctx->fontImages[i] = 0;
+		for (i = j; i < NVG_MAX_FONTIMAGES; i++) {
+			if (ctx->fontImages[i] != 0) {
+				nvgDeleteImage(ctx, ctx->fontImages[i]);
```

Comment:
I think this might currently remove too much. Imagine a case where we have 3 textures: A, B, C, and remove B. Then the array at this point will be A, C, C, and j=2 (after the remove it is A, 0, C, then C gets copied over). So we would delete C.

With the suggested change above, the array should be A, C, 0, after remove, and this loop is not needed at all.

Reply:
Good catch on the leak!